packages/tangle/tipmanager.go

```diff
@@ -192,117 +170,69 @@ func (t *TipManager) AddTip(message *Message) {
 		return
 	}
 
-	if !messageMetadata.IsEligible() {
-		return
-	}
-
-	if !t.tangle.ConsensusManager.PayloadLiked(messageID) {
-		return
-	}
-
 	// TODO: possible logical race condition if a child message gets added before its parents.
 	//  To be sure we probably need to check "It is not directly referenced by any strong message via strong/weak parent"
 	//  before adding a message as a tip. For now we're using only 1 worker after the scheduler and it shouldn't be a problem.
 
-	// if branch is monotonically liked: strong message
-	// if branch is not monotonically liked: weak message
+	if t.tips.Set(messageID, messageID) {
+		t.Events.TipAdded.Trigger(&TipEvent{
+			MessageID: messageID,
+		})
 
-	messageBranchID, err := t.tangle.Booker.MessageBranchID(messageID)
-	if err != nil {
-		// TODO: ALTERNATIVE ERROR HANDLING
-		panic(err)
+		t.tipsCleaner.ExecuteAt(messageID, func() {
+			t.tips.Delete(messageID)
+		}, message.IssuingTime().Add(tipLifeGracePeriod))
 	}
 
-	t.tangle.LedgerState.BranchDAG.Branch(messageBranchID).Consume(func(branch ledgerstate.Branch) {
-		if branch.MonotonicallyLiked() {
-			if t.strongTips.Set(messageID, messageID) {
-				t.Events.TipAdded.Trigger(&TipEvent{
-					MessageID: messageID,
-					TipType:   StrongTip,
-				})
-
-				t.tipsCleaner.ExecuteAt(messageID, func() {
-					t.strongTips.Delete(messageID)
-				}, message.IssuingTime().Add(tipLifeGracePeriod))
-			}
-
-			// skip removing tips if TangleWidth is enabled
-			if t.StrongTipCount()+t.WeakTipCount() <= t.tangle.Options.TangleWidth {
-				return
-			}
+	// skip removing tips if TangleWidth is enabled
+	if t.TipCount() <= t.tangle.Options.TangleWidth {
+		return
+	}
 
-			// a strong tip loses its tip status if it is referenced by a strong message via strong parent
-			message.ForEachParentByType(StrongParentType, func(parent MessageID) {
-				if _, deleted := t.strongTips.Delete(parent); deleted {
-					t.Events.TipRemoved.Trigger(&TipEvent{
-						MessageID: parent,
-						TipType:   StrongTip,
-					})
-				}
-			})
-			// a weak tip loses its tip status if it is referenced by a strong message via weak parent
-			message.ForEachParentByType(WeakParentType, func(parent MessageID) {
-				if _, deleted := t.weakTips.Delete(parent); deleted {
-					t.Events.TipRemoved.Trigger(&TipEvent{
-						MessageID: parent,
-						TipType:   WeakTip,
-					})
-				}
+	// a tip loses its tip status if it is referenced by another message
+	message.ForEachParent(func(parent Parent) {
```

Comment:
I think this should be `ForEachParentByType` with strong (and maybe weak, depends how we will ultimately use them) parents only.

Reply:
I changed it to strong parents for now.